test/java/seedu/addressbook/commands/ViewCommandTest.java

```diff
@@ -0,0 +1,125 @@
+        Person someone = new Person(new Name("me"),
+                                    new Phone("123", true),
+                                    new Email("some@hey.go", true),
+                                    new Address("nus", false),
+                                    new UniqueTagList(Collections.emptySet()));
+        listWithAll.add(someone);
+
+        // empty addressbook
+        assertViewErrorPersonNotInAddressBook(emptyAddressBook, listWithAll, 1);
+
+        // non-empty addressbook
+        assertViewErrorPersonNotInAddressBook(addressBook, listWithAll, listWithAll.size());
+    }
+
+    @Test
+    public void viewCommand_validIndex_returnsPersonDetails() {
+        // person with no private information
+        assertViewSuccess(addressBook, listWithAll, 1);
+
+        // person with some private information
+        assertViewSuccess(addressBook, listWithAll, 2);
+
+        // addressbook has more people than displayList
+        assertViewSuccess(addressBook, listWithSome, 1);
+    }
+
+    /**
+     * Creates a new view command.
+     */
+    private Command generateViewCommand(AddressBook addressBook, List<ReadOnlyPerson> displayList, int index) {
+        ViewCommand command = new ViewCommand(index);
+        command.setData(addressBook, displayList);
+
+        return command;
+    }
+
+    /**
+     * Asserts that the details person at specific index can be successfully retrieved
+     * and displayed.
+     */
+    private void assertViewSuccess(AddressBook addressBook, List<ReadOnlyPerson> list, int index) {
+        String expectedMessage = String.format(ViewCommand.MESSAGE_VIEW_PERSON_DETAILS,
+                                               list.get(index - 1).getAsTextHidePrivate());
+
+        Command command = generateViewCommand(addressBook, list, index);
+        assertCommandResult(command, expectedMessage, addressBook, TestUtil.clone(addressBook));
+    }
+
+    /**
+     * Asserts that the details of person at specific index cannot be retrieved due to
+     * invalid index.
+     */
+    private void assertViewErrorInvalidIndex(AddressBook addressBook, List<ReadOnlyPerson> list, int index) {
+        String expectedMessage = Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX;
+
+        Command command = generateViewCommand(addressBook, list, index);
+        assertCommandResult(command, expectedMessage, addressBook, TestUtil.clone(addressBook));
+    }
+
+    /**
+     * Asserts that the details of person at specific index cannot be retrieved due to
+     * person not existing in the addressbook.
+     */
+    private void assertViewErrorPersonNotInAddressBook(AddressBook addressBook, List<ReadOnlyPerson> list, int index) {
+        String expectedMessage = Messages.MESSAGE_PERSON_NOT_IN_ADDRESSBOOK;
+
+        Command command = generateViewCommand(addressBook, list, index);
+        assertCommandResult(command, expectedMessage, addressBook, TestUtil.clone(addressBook));
+    }
+}
```

Comment:
If this class is almost the same as the other test class, we can text both commands in one of the class and make the other class empty with just a comment to explain why it is empty.